realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -325,15 +331,31 @@ public static void setDefaultConfiguration(RealmConfiguration configuration) {
         if (configuration == null) {
             throw new IllegalArgumentException("A non-null RealmConfiguration must be provided");
         }
-        defaultConfiguration = configuration;
+        synchronized (monitorForDefaultConfiguration) {
+            defaultConfiguration = configuration;
+        }
+    }
+
+    /**
+     * Returns the default configuration for {@link #getDefaultInstance()}.
+     *
+     * @return default configuration object or {@code null} if {@link #init(Context)} was not called yet
```

Comment:
Smaller issue: As a general contract `Realm.init()` is required to be called before touching any other Realm API, so just throwing in this method if that hasn't been done would also be fine, and that is what most of our other methods do.